core/01_core.js

```diff
@@ -130,12 +130,21 @@
     return res;
   }
 
-  function opAsync(opName, arg1 = null, arg2 = null) {
+  function opAsync(opName, arg1 = null, arg2 = null, unref = false) {
     const promiseId = nextPromiseId++;
-    const maybeError = opcallAsync(opsCache[opName], promiseId, arg1, arg2);
+    const maybeError = opcallAsync(
+      opsCache[opName],
+      promiseId,
+      arg1,
+      arg2,
+      unref,
+    );
     // Handle sync error (e.g: error parsing args)
     if (maybeError) return unwrapOpResult(maybeError);
-    return PromisePrototypeThen(setPromise(promiseId), unwrapOpResult);
+    const p = PromisePrototypeThen(setPromise(promiseId), unwrapOpResult);
+    // Save the id on the promise so it can be latered ref'ed or unref'ed
+    p.id = promiseId;
```

Comment:
Have you considered using a symbol here or a reverse WeakMap?
(I am not sure what that would do for performance, we'd have to measure that)

Reply:
I did not, but using a symbol might be preferable, that way we can better communicate that users shouldn't use that value. I'll explore that.

Reply:
Used `Symbol.for("Deno.core.internalPromiseId")` instead, let's see if performance is better or worse